sktime/forecasting/compose/_multiplexer.py

```diff
@@ -0,0 +1,188 @@
+
+        Parameters
+        ----------
+        forecasters : list
+            List of (forecaster names, forecaster objects)
+            MultiplexForecaster switches between these forecasters
+            objects when used with ForecastingGridSearchCV to
+            find the optimal model
+        selected_forecaster: str
+            An argument to make a selection among forecasters.
+            MultiplexForecaster uses selected_forecaster
+            to choose which component to fit.
+            Important for using with ForecastingGridSearchCV as a
+            hyperparameter.
+
+        Attributes
+        ----------
+        _forecaster : Sktime forecaster
+            forecaster that MultiplexForecaster will currently
+            forecast with.
+        _forecaster_fit_params: dict
+            Fit params for the forecaster that is
+            going to be used by the forecaster in the
+            _forecaster
+        """
+
+        self.forecasters = forecasters
+        self.selected_forecaster = selected_forecaster
+        self._check_components()
+
+        super(MultiplexForecaster, self).__init__()
+
+    def _check_components(self):
+        if not isinstance(self.forecasters, list):
+            raise Exception(
+                "Please provide a list " "for forecasters composed of tuples"
+            )
+
+        for component in self.forecasters:
+            name, forecaster = component
+            if not isinstance(forecaster, _SktimeForecaster):
+                raise Exception(
+                    "Each component has to be an "
+                    "sktime forecaster object. "
+                    "Please check {} with name:{}".format(forecaster, name)
+                )
+
+    def _check_fit_params(self, fit_params):
+        forecaster_fit_params = {}
+
+        if self.selected_forecaster is None or fit_params == {}:
+            return forecaster_fit_params
+
+        for component in self.forecasters:
+            name, _ = component
+            if name not in fit_params.keys():
+                raise KeyError(
+                    "If you provide fit_params for models "
+                    "dictionary key of fit params need to "
+                    " match the associated component key."
+                )
+
+        if self.selected_forecaster in fit_params.keys():
+            forecaster_fit_params = fit_params[self.selected_forecaster]
+            return forecaster_fit_params
+        else:
+            return forecaster_fit_params
+
+    def _check_selected_forecaster(self):
+        component_names = [name for name, _ in self.forecasters]
+        if self.selected_forecaster not in component_names:
+            raise Exception(
+                "Please check the selected_forecaster argument provided "
+                " Valid selected_forecaster parameters: {}".format(component_names)
+            )
+
+    def _update_forecaster(self):
```

Comment:
```suggestion
    def _set_forecaster(self):
```
to avoid ambiguity with `update`